nixos/modules/services/web-apps/jitsi-meet.nix

```diff
@@ -0,0 +1,362 @@
+      type = attrs;
+      default = { };
+      example = literalExample ''
+        {
+          SHOW_JITSI_WATERMARK = false;
+          SHOW_WATERMARK_FOR_GUESTS = false;
+        }
+      '';
+      description = ''
+        Client-side web-app interface settings that override the defaults in <filename>interface_config.js</filename>.
+
+        See <link xlink:href="https://github.com/jitsi/jitsi-meet/blob/master/interface_config.js" /> for
+        default configuration with comments.
+      '';
+    };
+
+    videobridge.enable = mkOption {
+      type = bool;
+      default = true;
+      description = ''
+        Whether to enable Jitsi Videobridge instance and configure it to connect to Prosody.
+
+        Additional configuration is possible with <option>services.jitsi-videobridge</option>.
+      '';
+    };
+
+    jicofo = {
+      config = mkOption {
+        type = attrsOf str;
+        default = { };
+        example = literalExample ''
+          {
+            "org.jitsi.jicofo.auth.URL" = "XMPP:jitsi-meet.example.com";
+          }
+        '';
+        description = ''
+          Contents of the <filename>sip-communicator.properties</filename> configuration file for jicofo.
+        '';
+      };
+    };
+
+    nginx.enable = mkOption {
+      type = bool;
+      default = true;
+      description = ''
+        Whether to enable nginx virtual host that will serve the javascript application and act as
+        a proxy for the XMPP server. Further nginx configuration can be done by adapting
+        <option>services.nginx.virtualHosts.&lt;hostName&gt;</option>. It is highly recommended to
+        enable the <option>enableACME</option> and <option>forceSSL</option> options:
+
+        <programlisting>
+        services.nginx.virtualHosts.''${config.services.jitsi-meet.hostName} = {
+          enableACME = true;
+          forceSSL = true;
+        };
+        </programlisting>
+      '';
+    };
+
+    prosody.enable = mkOption {
+      type = bool;
+      default = true;
+      description = ''
+        Whether to configure Prosody to relay XMPP messages between Jitsi Meet components. Turn this
+        off if you want to configure it manually.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.jitsi-meet.jicofo.config = mapAttrs (_: v: mkDefault v) {
+      "org.jitsi.jicofo.BRIDGE_MUC" = "jvbbrewery@internal.${cfg.hostName}";
+      "org.jitsi.jicofo.ALWAYS_TRUST_MODE_ENABLED" = "true";
+    };
+
+    services.prosody = mkIf cfg.prosody.enable {
+      enable = mkDefault true;
+      modules = {
+        admin_adhoc = mkDefault false;
+        bosh = mkDefault true;
+        ping = mkDefault true;
+        roster = mkDefault true;
+        saslauth = mkDefault true;
+        tls = mkDefault true;
+      };
+      extraModules = [ "pubsub" ];
+      extraConfig = ''
+        certificates = "${config.services.prosody.dataDir}"
+
+        Component "conference.${cfg.hostName}" "muc"
```

Comment:
Can these component names be like in their [official documentation](https://github.com/jitsi/jitsi-meet/blob/master/doc/manual-install.md)?

Reply:
They are - `conference.${cfg.hostName}` and `focus.${cfg.hostName}` are the same.

The linked document describes connecting videobridge as a XMPP *component* named `jitsi-videobridge.${cfg.hostName}`, which is deprecated. It is now recommended to connect the videobridge as a XMPP *client* that joins the `internal.${cfg.hostName}` MUC component.

See this part of the official [DEB package](https://github.com/jitsi/jitsi-meet/blob/master/debian/jitsi-meet-prosody.postinst#L136) or their [docker](https://github.com/jitsi/docker-jitsi-meet/blob/master/prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua#L85), [repo](https://github.com/jitsi/docker-jitsi-meet/blob/master/env.example#L189).

Reply:
Also regarding deprecation: https://github.com/jitsi/jitsi-videobridge/blob/master/doc/muc.md

Reply:
So [manual-install](https://github.com/jitsi/jitsi-meet/blob/master/doc/manual-install.md), updated 20 days ago is documenting a deprecated installation method? Huuuh…

Reply:
@flokli 

> WARNING: Manual installation is not recommended. We recommend following the quick-install document. The current document describes the steps that are needed to install a working deployment, but steps are easy to mess up, and **the debian packages are more up-to-date, where this document is sometimes not updated to reflect latest changes.**

Seems to me like the Debian package should be authoritative, and this PR matches what it does.

Reply:
Unfortunately the documentation is a bit lacking and needs to be combined with reverse engineering the Debian packages and digging through [Jitsi Discourse](https://community.jitsi.org/):(

For example according to https://github.com/jitsi/jitsi-meet/blob/master/doc/scalable-installation.md#jitsi-videobridge2 manual configuration of NAT addresses is no longer needed which I had no idea about.